src/web/tools/GestureHandlerOrchestrator.ts

```diff
@@ -73,13 +73,24 @@ export default class GestureHandlerOrchestrator {
     if (this.hasOtherHandlerToWaitFor(handler)) {
       this.addAwaitingHandler(handler);
     } else if (
+      this.shouldActivate(handler) &&
```

Comment:
Have you checked if handlers not activated because of this check are being correctly canceled/are failing?

Reply:
Good point, they should fail/be canceled right away.